bijection-thrift/src/main/scala/com/twitter/bijection/thrift/ThriftCodecs.scala

```diff
@@ -63,7 +63,23 @@ object BinaryThriftCodec {
 }
 
 class BinaryThriftCodec[T <: TBase[_, _]](klass: Class[T])
-  extends ThriftCodec[T, TBinaryProtocol.Factory](klass, new TBinaryProtocol.Factory)
+  extends Injection[T, Array[Byte]] {
+
+  private[this] val factory = new TBinaryProtocol.Factory
+
+  protected lazy val prototype = klass.newInstance
+
+  override def apply(item: T) = {
+    val baos = new ByteArrayOutputStream
+    item.write(factory.getProtocol(new TIOStreamTransport(baos)))
+    baos.toByteArray
+  }
+  override def invert(bytes: Array[Byte]) = attempt(bytes) { bytes =>
+    val obj = prototype.deepCopy
```

Comment:
So thats a good question, not sure why its always deep copy. I took it from the existing code. Any idea if there was a bug/functional reason between the two?